### ISSUE TITLE
Reopening: TypeError: Cannot convert a BigInt value to a number #771

### ISSUE THREAD

**Issue by wmaillard** [amazon-ion/ion-js#773]
I'm getting the same error while trying to import [this file](https://amazon-ion.github.io/ion-js/browser/scripts/ion-bundle.min.js) into a tampermonkey script.

### Steps to reproduce

1. Open [the codepen](https://codepen.io/robcopeland/) in Chrome (I'm on 122.0.6261.112)
2. Right click, inspect
3. Look at the **Dev tools console** (not the codepen console)

```
ion-bundle.js:264 Uncaught TypeError: Cannot convert a BigInt value to a number
    at Math.pow (<anonymous>)
    at Function.calculateSizeThreshold (ion-bundle.js:264:28)
    at Function.calculateSizeThresholds (ion-bundle.js:256:30)
    at 2.@babel/runtime/helpers/classCallCheck (ion-bundle.js:274:43)
    at o (ion-bundle.js:1:628)
    at ion-bundle.js:1:679
    at 15../BigIntSerde (ion-bundle.js:2608:21)
    at o (ion-bundle.js:1:628)
    at ion-bundle.js:1:679
    at 7../IntSize (ion-bundle.js:690:28)
```

_Originally posted by @wmaillard in https://github.com/amazon-ion/ion-js/issues/771#issuecomment-2024330612_

**Comment by wmaillard** [comment]
Here's the problem, [the code here](https://github.com/amazon-ion/ion-js/blob/ebd89660626e5374489d0ff7bdc1d9875845f2f3/src/BigIntSerde.ts#L122) has 
```2n ** exponent ``` (when I set a breakpoint, exponent is 8n)

If you open you chrome devtools console and paste in
```2n ** 8n``` the result is 256n

I'm guessing, is that when the TS is compiled, ```2n ** exponent ``` gets replaced with ```Math.pow(2n, exponent)``` -> search for this Math.pow code in [the compiled js](https://amazon-ion.github.io/ion-js/browser/scripts/ion-bundle.js)

You'll see Math.pow(2n, exponent) is in there right where 2n ** exponent should be.

**Comment by wmaillard** [comment]
Here's your problem:
https://github.com/babel/babel/issues/13109

**Comment by desaikd** [comment]
Hi @wmaillard, We really appreciate you taking the time to report an issue. I was able to see the issue with steps your provided in codepen. I am working on reproducing it locally with `ion-js` library and then create a PR to fix this by following the solution pointed here: https://github.com/babel/babel/issues/13109#issuecomment-826287089.

**Comment by wmaillard** [comment]
@desaikd thanks merging the change.  I'm looking at [this file](https://amazon-ion.github.io/ion-js/browser/scripts/ion-bundle.js) and wondering if it has been updated with the changes (I still see `Math.pow(2n, exponent)`)?  I looked a little at your deployments but don't know if I'm looking in the correct spot, the [node-js workflow](https://github.com/amazon-ion/ion-js/actions/workflows/build.yml) succeeded, but the [ion-js workflow](https://github.com/amazon-ion/ion-js/actions/workflows/release.yml) hasn't been touched in a while.

**Comment by desaikd** [comment]
The release for this change is till pending which is why `ion-js release` workflow doesn't show anything yet. The `node-js workflow` is run for each PR and succeeded like you said. I will update this issue once a new release is out.

**Comment by desaikd** [comment]
Closing this as the fix has been released with v5.2.1: https://github.com/amazon-ion/ion-js/releases/tag/v5.2.1. Feel free to reopen if the issue persists or if you have any additional comments.
